dev_guide/configmaps.adoc

```diff
@@ -339,39 +329,46 @@ spec:
     - name: test-container
       image: gcr.io/google_containers/busybox
       command: [ "/bin/sh", "-c", "env" ]
-      env:
+      env: <1>
         - name: SPECIAL_LEVEL_KEY
           valueFrom:
-            configMapKeyRef:
-              name: special-config
-              key: special.how
+            configMapKeyRef: 
+              name: special-config <2>
+              key: special.how <3>
         - name: SPECIAL_TYPE_KEY
           valueFrom:
-            configMapKeyRef:
-              name: special-config
-              key: special.type
+            configMapKeyRef: 
+              name: special-config <2>
+              key: special.type <3>
+              optional: true <4>
+        envFrom:<5>
+          - configMapRef:
+              name: env-config <6>
   restartPolicy: Never
 ----
+
+<1> Stanza to pull specific environment variables from a ConfigMap.
+<2> Name of the ConfigMap to pull specific environment variables.
+<3> Environment variable to pull from the ConfigMap. 
+<4> Marks the environment variable as optional and the variable will not be used.
```

Comment:
"...as optional; the pod will be started even if this configmap and key do not exist"